File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const { log } = require("console");
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = Number(input[0]);
let dp = new Array(11);

dp[1] = 1;
dp[2] = 2;
dp[3] = 4;

for (let i = 4; i < 11; i++) {
  dp[i] = dp[i - 3] + dp[i - 2] + dp[i - 1];
}

for (let i = 1; i <= n; i++) {
  let t = Number(input[i]);
  console.log(dp[t]);
}

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
t = int(input())
for _ in range(t):
    n = int(input())
    dp = [0] * n

    for i in range(n):
        if i == 0:
            dp[i] = 1
        elif i == 1:
            dp[i] = 2
        elif i == 2:
            dp[i] = 4
        else:
            dp[i] = dp[i-3] + dp[i-2] + dp[i-1]

    print(dp[n-1])